pkg/runtime/start.go

```diff
@@ -0,0 +1,45 @@
+package runtime
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"golang.org/x/sync/errgroup"
+)
+
+func (r *Runtime) Start() {
```

Comment:
This func could return an error and the caller can handle it instead of logging at this level.
This is subjective though so I'll leave you to make the call.

Reply:
I was thinking about this, the error would then need to be handled at the `/cmd` level which is why I took this approach. I would be keen to see if anyone feels strongly either way

Reply:
I suppose given the naming of `cmd` vs `runtime` I think handling the errors at the level you've chosen makes sense